app/services/organizations/importers/google_csv_import_service.rb

```diff
@@ -13,42 +13,57 @@ def initialize(file)
-          previously_matched_form_submission = FormSubmission.where(person:, csv_timestamp:)
+          CSV.foreach(@file.to_path, headers: true, skip_blanks: true).with_index(1) do |row, index|
+            # Using Google Form headers
+            email = row["Email"].downcase
+            csv_timestamp = Time.parse(row["Timestamp"])
 
-          if person.nil?
-            @no_match << [index, email]
-          elsif previously_matched_form_submission.present?
-            next
-          else
-            latest_form_submission = person.latest_form_submission
-            ActiveRecord::Base.transaction do
-              # This checks for the empty form submission that is added when a person is created
-              if latest_form_submission.csv_timestamp.nil? && latest_form_submission.form_answers.empty?
-                latest_form_submission.update!(csv_timestamp:)
-                create_form_answers(latest_form_submission, row)
-              else
-                # if the person submits a new/updated form,
-                # i.e. an additional row in the csv with the same email / different timestamp,
-                # a new form_submission will be created
-                create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
+            person = Person.find_by(email:, organization: @organization)
+            previously_matched_form_submission = FormSubmission.where(person:, csv_timestamp:)
+
+            if person.nil?
+              @no_match << [index, email]
+            elsif previously_matched_form_submission.present?
+              next
+            else
+              latest_form_submission = person.latest_form_submission
+              ActiveRecord::Base.transaction do
+                # This checks for the empty form submission that is added when a person is created
+                if latest_form_submission.csv_timestamp.nil? && latest_form_submission.form_answers.empty?
+                  latest_form_submission.update!(csv_timestamp:)
+                  create_form_answers(latest_form_submission, row)
+                else
+                  # if the person submits a new/updated form,
+                  # i.e. an additional row in the csv with the same email / different timestamp,
+                  # a new form_submission will be created
+                  create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
+                end
+                @count += 1
               end
-              @count += 1
             end
+          rescue => e
+            @errors << [index, e]
           end
-        rescue => e
-          @errors << [index, e]
         end
         Status.new(@errors.empty?, @count, @no_match, @errors)
       end
 
       private
 
+      def validate_file
+        raise FileTypeError unless @file.content_type == "text/csv"
+
+        first_row = CSV.foreach(@file.to_path).first
+        raise EmailColumnError if first_row.nil?
```

Comment:
If I am understanding this correctly, we are raising `EmailColumnError` if the first row is empty, right? Should we have a different error class to handle this case? Something like `NoDataError` (I am sure there is a better name than that).

Reply:
Yes it is checking for an empty file as an edge case (came up during testing). I just reused the same error because technically the file does not have the "Email" column but I can add a more specific error.